mattvryan/jackrabbit-oak
# OAK-7569 - improved test code, cover getProperty().getBinary(), fix multipart test

With this change, many tests will fail expectedly (until the ValueFactoryImpl problem is fixed):

```
[INFO] Running org.apache.jackrabbit.oak.jcr.binary.BinaryAccessIT
[ERROR] Tests run: 136, Failures: 12, Errors: 32, Skipped: 4, Time elapsed: 100.639 s <<< FAILURE! - in org.apache.jackrabbit.oak.jcr.binary.BinaryAccessIT
```

Test fixes:
* ensure `getProperty().getBinary()` code path is tested (currently broken)
* multipart test did not really do a multipart upload, adapted it according to the upload algorithm in the BinaryUpload javadoc

Test improvements:
* check fixture.isAvailable() in AbstractRepositoryTest to get "skipped" junit reports if S3 or Azure fixture is not configured instead of silently dropping them
* add support for reusable tests in AbstractRepositoryTest instead of hacking it into the fixtures
* consistent `BinaryAccess*` class naming pattern
* separated out DataStoreFixture and implementations as top-level classes
* introduced `Content` utility to simplify binary content test code
